pywatts_pipeline/core/pipeline.py

```diff
@@ -57,16 +60,110 @@ class Pipeline(BaseTransformer):
     :type path: str
     """
 
-    def __init__(self, path: Optional[str] = ".", name="Pipeline"):
+    def __init__(self, path: Optional[str] = ".", name="Pipeline", steps=None):
```

Comment:
The constructor API needs to look differently. We need to distinguish between structure and modules here, since sklearn assumes that if parameters are deep cloned stay the same. However, in case if a module is added multiple times to the pipeline, there are two entries in the list steps that should contain the identical module. Howevever because of the clone they are not identical any more